execution/limit.go

```diff
@@ -68,19 +71,32 @@ func (node *LimitedStream) Close() error {
 }
 
 func (node *LimitedStream) Next(ctx context.Context) (*Record, error) {
-	if node.limit > 0 {
+	tx := storage.GetStateTransactionFromContext(ctx)
+	limitState := storage.NewValueState(tx.WithPrefix(node.streamID.AsPrefix()).WithPrefix(limitPrefix))
+
+	var limit octosql.Value
+	err := limitState.Get(&limit)
+	if err != nil {
+		return nil, errors.Wrap(err, "couldn't get limit state")
+	}
+
+	if limit.AsInt() > 0 {
 		record, err := node.rs.Next(ctx)
 		if err != nil {
 			if err == ErrEndOfStream {
-				node.limit = 0
+				limit = octosql.MakeInt(0)
 				return nil, ErrEndOfStream
 			}
 			return nil, errors.Wrap(err, "LimitedStream: couldn't get record")
```

Comment:
wrap body starts with Uppercase

Reply:
same with wrap inside Close method

Reply:
That was like that previously, but yeah, I'll change it.